File: movie_merge/movie_merge/track.rb
require 'simple_time'

class Track
  attr_accessor :duration, :title

  def initialize(duration, title)
  	@duration = duration
  	@title = title
  end

  def self.parse(str)
  	results = str.split(",")
  	duration_str = results[0]
  	title_str = results[1]
  	Track.new(SimpleTime.parse(duration_str), title_str)
  end

end

File: scala_movie_merge/src/main/scala/com/eaw/moviemerge/Track.scala
package com.eaw.moviemerge

/**
  * Created by erikwenberg on 12/25/17.
  */
class Track(val duration: SimpleTime, val title: String) {

}

object Track {
  def parse(s: String): Track = {
    val parts = s.split(",")
    new Track(SimpleTime.parse(parts(0)), parts(1))
  }
}
